AWS PyCharm Tutorial Series (#155)
JetBrains/jetbrains_guide: sites/pycharm-guide/contents/tutorials/intro-aws/deploying/index.md

---
type: TutorialStep
date: 2021-01-29
title: Deploying Functions
technologies: [AWS Toolkit]
topics: [aws]
author: mm

> subtitle: Deploying Lambda APIs through Pycharm

Reviewer: ```suggestion
subtitle: Use PyCharm Professional to deploy your Lambda APIs.
```

thumbnail: ../thumbnail.png
longVideo:
  poster: ./poster_long.png
  url: https://youtu.be/mqNTLSOcSOw
---

In this tutorial we are going to deploy our functions in AWS and test it through Postman.

# Beginning Launch in AWS

![deploy_step_1](./steps/step1.png)

Before we start, I want to give some information, as you can see in the above image,
I have chosen <strong>AdministratorAccess</strong> which provides complete access to
all the AWS Resources. In my previous video, I used different policies, there is no
problem in that, but recently there was information shared by AWS
that AWS Lambda will deprecate two managed policies called <strong>"AWSLambdaFullAccess"</strong> 
and <strong>"AWSLambdaReadOnlyAccess"</strong> on January 25, 2021.

For this tutorial I have used <strong>AdministratorAccess</strong>, but to be honest

> its not recommended approach from point of security and access controls.

Reviewer: ```suggestion
it's not the recommended approach from point of security and access controls.
```

You can see the list of APIs along-with their request Methods like GET, POST, PUT and DELETE.

![deploy_step_7](./steps/step7.png)

Next, I will click on <strong>Stages</strong> which you can see in the left sidebar. 
You remember we created the stage name as <strong>Prod</strong> in the <strong>template.yaml</strong> file.

![deploy_step_8](./steps/step8.png)

You can see the <strong>Invoke URL</strong> in the above image, this the URL from which we are going to access our API, 
I am going to copy the URL and paste it in Postman.

Okay, let’s begin by testing the <strong>Login API</strong>. I am going to provide
email and password. If the credentials are authenticated successfully then
I will receive a [JWT](https://jwt.io/) token.

![deploy_step_9](./steps/step9.png)

As you can see I have been successfully authenticated and also I received a token in response.

This token will be used subsequently in other operations to authorize the requests.

I am going to retrieve list of users, for that I need to pass the token in the <strong>Authorization</strong> header,

> and the request method is got to be <strong>GET</strong>. 

Reviewer: ```suggestion
and the request method must be <strong>GET</strong>. 
```

You can see I am able to retrieve all users. 

![deploy_step_10](./steps/step10.png)

Let me try to get information for a specific user. You can see that I am also receiving information for that particular user.

![deploy_step_11](./steps/step11.png)

Now, I am going to create a new user, and the request method is going to be <strong>POST</strong>. I will be providing all the necessary information, make sure to pass the token in the Authorization header.

![deploy_step_12](./steps/step12.png)

> Okay, we have successfully created a new user. So, I am going to perform update operation on the same record.

Reviewer: ```suggestion
Okay, we have successfully created a new user. So, I am going to perform an update operation on the same record.
```
